main.tf

```
@@ -36,12 +55,34 @@ resource "aws_transfer_ssh_key" "default" {
   ]
 }
 
+# Custom Domain
+resource "aws_route53_record" "main" {
+  count   = length(var.domain_name) > 0 && length(var.zone_id) > 0 ? 1 : 0
```

This should be disabled if the module is disabled. The `local.enabled` should be equal to `module.this.enabled` at the top.

```suggestion
  count   = local.enabled && length(var.domain_name) > 0 && length(var.zone_id) > 0 ? 1 : 0
```